src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +452,317 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
                 iEndIndex = endIndex
             };
 
-            return expansion.InsertExpansion(textSpan, textSpan, this, LanguageServiceGuid, out ExpansionSession) == VSConstants.S_OK;
+            if (expansion.InsertExpansion(textSpan, textSpan, this, LanguageServiceGuid, out _state._expansionSession) == VSConstants.S_OK)
+            {
+                // This expansion is not derived from a symbol
+                _state.ClearSymbolInformation();
+                return true;
+            }
+
+            if (!(SubjectBuffer.GetFeatureOnOffOption(CompletionOptions.EnableArgumentCompletionSnippets) ?? false))
+            {
+                // Argument completion snippets are not enabled
+                return false;
+            }
+
+            var symbols = ThreadingContext.JoinableTaskFactory.Run(() => GetSymbolsAsync(caretPosition: triggerSpan.End, cancellationToken));
```

Comment:
in general, what is trigger span?  is it the end of a method name if hte user types something like `WriteLine<tab><tab>`?

Reply:
It's the identifier to the left of the caret.